torch/csrc/jit/codegen/cuda/manager.cpp

```diff
@@ -49,29 +61,52 @@ class CudaFusionManager {
 
       graph_cache_[repr] = kernel_id;
 
-      Fusion fusion;
-      // lower torch::jit::Graph to torch::jit::fuser::cuda::fusion
-      parseJitIR(graph, fusion);
-
-      // default constructor via accessing empty key;
-      compileKernel(fusion, kernel_cache_[kernel_id]);
+      // create entry for cached kernel;
+      kernel_cache_.insert({kernel_id, CudaKernelCache()});
 
-      return kernel_id;
-    } else {
-      return graph_cache_[repr];
+      // TODO: we should compile here using profiled information:
+      //       size (range) / stride (contiguity)
     }
+
+    return graph_cache_[repr];
   };
 
   void runFusionNode(
       int32_t kernel_id,
+      std::shared_ptr<Graph>& graph,
       const at::ArrayRef<IValue> inputs,
       std::vector<at::Tensor> outputs) {
-    TORCH_CHECK(
-        kernel_cache_.count(kernel_id) != 0, "kernel id not recognized");
+    std::lock_guard<std::mutex> guard(mutex_);
```

Comment:
What's the reason to guard the kernel cache? Are we expecting multiple threads to be executing JIT code? Codegen is unlikely thread safe, do we need it to be?

Reply:
I was just being paranoid.
if multithreading is not a concern I'll just remove it.

Reply:
I honestly don't know if it is :-/ we should probably follow up with Zach and ask.